### ISSUE TITLE
Wrong dead code analysis on pattern with bool

### ISSUE THREAD

**Issue by ChaserVasya** [dart-lang/sdk#56222]
# Snippet
```
void main() {
  const boolContainer = (b: true);
  if (true || boolContainer.b) {}
}

```
# Problem
`{}` is recognized as dead code. It is wrong
<img width="330" alt="Снимок экрана 2024-07-11 в 13 08 53" src="https://github.com/dart-lang/sdk/assets/74578917/ded0c399-701a-4665-b16b-9445a8c53b7a">

# dart info
- Dart 3.2.3 (stable) (Tue Dec 5 17:58:33 2023 +0000) on "macos_arm64"
- on macos / Version 14.5 (Build 23F79)
- locale is ru-RU

**Comment by srawlins** [comment]
Yeah the highlight range seems to include the `{` and anything else to the right, on the line with `{`. Bad range.
